library/servers/servers.c

```diff
@@ -45,3 +45,26 @@ int file_read(int file_ino, uint offset, char* block) {
 
     return reply->status == FILE_OK? 0 : -1;
 }
+
+int term_read(char *buf, uint len) {
+    if (grass->print_status == PRINT_KERNEL) FATAL("term_read: kernel reading chars through server interface");
```

Comment:
See comments for `term_write` in `servers.c`

Reply:
Removed crash in `term_read` for the Kernel, as `KERNEL` is defined within `sys_shell`, so `term_read` would crash for the Shell. Let me know if you believe that we should add a case for the OS calling `term_read`.